### PR TITLE
ATT-309: implement encryption and erasure coding algorithm

### PR DESCRIPTION
**Pull request by wilsonangara**
While at it, also created the UI to show whether the the file is a valid fhir or not

### CODE REVIEW

File: app/(protected)/dashboard/(admin)/patient/accounts/form/ConfirmUploadedEMR.tsx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+	const renderResult = (result: boolean) => {
+		if (result) {
+			return <Typography>Valid</Typography>;
+		}
+		return <Typography sx={{ color: "red" }}>Invalid</Typography>;
+	};
+
+	// renderIcon is used for generated checked or unchecked icon for
+	// both `matchedVirtualAccount` and `fileUploaded` field.
+	const renderIcon = (value: boolean) => {
```
Comment (linerocks) [inline app/(protected)/dashboard/(admin)/patient/accounts/form/ConfirmUploadedEMR.tsx:43]: Move it to a separate component

File: app/(protected)/dashboard/(admin)/patient/accounts/form/ConfirmUploadedEMR.tsx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+export const ConfirmUploadedEMR: React.FC = () => {
+	const { isLoading, files } = useSelector(
+		(state: RootState) => state.uploadEMR
+	);
+
+	const renderMetadata = () => {
+		return <Typography>Data</Typography>;
+	};
+
+	const renderResult = (result: boolean) => {
```
Comment (linerocks) [inline app/(protected)/dashboard/(admin)/patient/accounts/form/ConfirmUploadedEMR.tsx:34]: Move it to a separate component

File: app/(protected)/dashboard/(admin)/patient/accounts/form/ConfirmUploadedEMR.tsx [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+	fontWeight: 600,
+	marginTop: "50px"
+}));
+
+export const ConfirmUploadedEMR: React.FC = () => {
+	const { isLoading, files } = useSelector(
+		(state: RootState) => state.uploadEMR
+	);
+
+	const renderMetadata = () => {
```
Comment (linerocks) [inline app/(protected)/dashboard/(admin)/patient/accounts/form/ConfirmUploadedEMR.tsx:30]: Move it to a separate component

File: app/(protected)/dashboard/(admin)/patient/accounts/form/UploadEMR.tsx [hunk 4]
```diff
@@ -69,45 +77,90 @@ export const UploadEMR: React.FC = () => {
...
+							index: i,
+							signature: hashedFragment,
+							value: util.encodeBase64(fragment)
+						};
+					}
+				);
+				splittedChunks.push({
+					value: chunk,
+					hash: hashedChunk,
+					framgents: formattedFragments
```
Comment (linerocks) [inline app/(protected)/dashboard/(admin)/patient/accounts/form/UploadEMR.tsx:142]: typo

File: app/(protected)/dashboard/(admin)/patient/accounts/form/ConfirmUploadedEMR.tsx (hunk 1 above)
Comment (wilsonangara) [reply app/(protected)/dashboard/(admin)/patient/accounts/form/ConfirmUploadedEMR.tsx:43]: do you mean creating a seperate component to return the icons?

File: app/(protected)/dashboard/(admin)/patient/accounts/form/ConfirmUploadedEMR.tsx (hunk 3 above)
Comment (linerocks) [reply app/(protected)/dashboard/(admin)/patient/accounts/form/ConfirmUploadedEMR.tsx:30]: const MetadataLabel: React.FC<MetadataLabelInterface> = (props) =>{
   return <Typography>Data</Typography>;
}

**Review by altairrostam — APPROVED** [review]
LGTM

**Pull request merged**